### PR TITLE
Creating big query tables for perf tests - 3

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Changing python script, to upload data in big query tables(Added functions). 
Added licence files.

ref - https://github.com/GoogleCloudPlatform/gcsfuse/pull/1209

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Tested build script manually, Working fine uploading data on gsheet.
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def setup_dataset_and_tables(self):
       CREATE TABLE IF NOT EXISTS {}.{}.{}(
         configuration_id STRING,
         start_time_build INT64,
+        mount_type STRING,
```
Comment (Tulsishah) [inline perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py:215]: For FYI @raj-prince 
This is for pd MOUNT_TYPE_GCS = 'gcs_bucket' MOUNT_TYPE_PD = 'persistent_disk'

File: perfmetrics/scripts/vm_metrics/vm_metrics.py [hunk 2]
```diff
@@ -312,7 +326,9 @@ def fetch_metrics(self, start_time_sec, end_time_sec, instance, period, test_typ
       row.append(end_time_sec)
       for metric in updated_metrics_list:
         row.append(metric.metric_point_list[i].value)
-      metrics_data.append(row)
+
+      # Skipping the first column as it duplicates the second column.
+      metrics_data.append(row[1:])
```
Comment (Tulsishah) [inline perfmetrics/scripts/vm_metrics/vm_metrics.py:331]: FYI @raj-prince 
Ruchika had a [logic](https://github.com/GoogleCloudPlatform/gcsfuse/blob/experiments_dashboard/perfmetrics/scripts/vm_metrics/vm_metrics.py#L325) to append None values at the end, which seems unnecessary to me, as in our current code also we are not following the logic

File: perfmetrics/scripts/fetch_metrics.py [hunk 3]
```diff
@@ -51,15 +88,23 @@ def _parse_arguments(argv):
...
   args = _parse_arguments(argv)
 
+  temp = fio_metrics_obj.get_metrics(args.fio_json_output_path)
+  metrics_data = fio_metrics_obj.get_values_to_upload(temp)
+
   if args.upload_gs:
-    temp = fio_metrics_obj.get_metrics(args.fio_json_output_path, FIO_WORKSHEET_NAME)
-  else:
-    temp = fio_metrics_obj.get_metrics(args.fio_json_output_path)
+    gsheet.write_to_google_sheet(metrics_data, FIO_WORKSHEET_NAME)
```
Comment (Tulsishah) [inline perfmetrics/scripts/fetch_metrics.py:95]: FYI @raj-prince 
Ruchika created one other function named [upload_metrics_to_gsheet](https://github.com/GoogleCloudPlatform/gcsfuse/blob/experiments_dashboard/perfmetrics/scripts/fio/fio_metrics.py#L462), which is nothing but making unnecessary cover on [write_to_google_sheet ](https://github.com/GoogleCloudPlatform/gcsfuse/blob/experiments_dashboard/perfmetrics/scripts/gsheet/gsheet.py#L18). So I didn't copy that function and corresponding unit tests in fio_metrics.py file

File: perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py (hunk 1 above)
Comment (raj-prince) [reply perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py:215]: Do we use this field in the current setup? If not then we should remove this.

File: perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh [hunk 4]
```diff
@@ -30,6 +30,9 @@ chmod +x perfmetrics/scripts/build_and_install_gcsfuse.sh
 # Mounting gcs bucket
 cd "./perfmetrics/scripts/"
 
+echo Installing requirements..
```
Comment (raj-prince) [inline perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh:33]: let's be specific - Installing Bigquery module requirements...

File: perfmetrics/scripts/fio/fio_metrics.py [hunk 5]
```diff
@@ -422,31 +440,32 @@ def _add_to_gsheet(self, jobs, worksheet_name):
...
       List of dicts, contains list of jobs and required metrics for each job
     """
     fio_out = self._load_file_dict(filepath)
     job_metrics = self._extract_metrics(fio_out)
-    if worksheet_name:
-      self._add_to_gsheet(job_metrics, worksheet_name)
-
     return job_metrics
 
+  def upload_metrics_to_bigquery(self, metrics_data, config_id, start_time_build, table_id_bq):
```
Comment (raj-prince) [inline perfmetrics/scripts/fio/fio_metrics.py:458]: Let's keep the input parameter in same order - table_id, config_id, start_time_build, metrics_data

File: perfmetrics/scripts/vm_metrics/vm_metrics.py (hunk 2 above)
Comment (raj-prince) [reply perfmetrics/scripts/vm_metrics/vm_metrics.py:331]: As we discussed offline, that was being added in the last for different fields, but skipping logic is required.

**Review by raj-prince — CHANGES_REQUESTED** [review]
Few minor comments.

File: perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py (hunk 1 above)
Comment (Tulsishah) [reply perfmetrics/scripts/bigquery/experiments_gcsfuse_bq.py:215]: Yes, we are

**Comment by Tulsishah** [comment]
> Few minor comments.

Fixed and tested again.

**Review by raj-prince — APPROVED** [review]

**Comment by gargnitingoogle** [comment]
@mustvicky@gmail.com test

**Comment by mustvicky** [comment]
Confirmed to let the commit go to Master. My understanding is that if there is a problem this causes in regression test, we should rollback this chaneg to not release a possibly incorrect software, even if the software is not in the main logic. 

Also, in general, we should tag master before runnign the longhaul tests release the build from that tag. I understand we are not there yet.

**Pull request merged**